### PR TITLE
Dockerize cirque build

### PR DESCRIPTION
**Pull request by rwalker-apple**
#### Problem
 Cirque scripts assume apt
 * installs system deps on dev's box
 * doesn't work on MacOS

 #### Summary of Changes
 move system dependencies to a Docker image, use the image in CI

with this image in place, I can reproduce Cirque CI locally on MacOS with:
```
$ integrations/docker/images/chip-build-cirque/run.sh  scripts/tests/cirque_tests.sh ...
```

### CODE REVIEW

**Comment by rwalker-apple** [comment]
@erjiaqing any issues?

**Comment by erjiaqing** [comment]
> @erjiaqing any issues?

In the future, cirque test will run wifi / ble tests, which requires kernel modules / bluez simulation, they does not support macos etc.

It will be fine if we just run thread tests (and data model tests via thread)

@yunhanw-google

**Comment by rwalker-apple** [comment]
> > @erjiaqing any issues?
> 
> In the future, cirque test will run wifi / ble tests, which requires kernel modules / bluez simulation, they does not support macos etc.
> 
> It will be fine if we just run thread tests (and data model tests via thread)
> 
> @yunhanw-google

The goal here is to support MacOS with a Linux container, a la github's ubuntu-latest, but with an image we supply.

**Comment by rwalker-apple** [comment]
@erjiaqing @yunhanw-google can you have a look at https://github.com/project-chip/connectedhomeip/suites/1258803035/artifacts/19254576 and help me make sense of it?

File: integrations/docker/images/chip-build-cirque/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Docker connectedhomeip:chip-build
```
Comment (bzbarsky-apple) [inline integrations/docker/images/chip-build-cirque/README.md:1]: This documentation doesn't seem chip-build-cirque-specific.  Was it supposed to go in some ancestor dir?

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: integrations/docker/images/chip-build-cirque/README.md (hunk 1 above)
Comment (rwalker-apple) [reply integrations/docker/images/chip-build-cirque/README.md:1]: copy-pasta'd from chip-build/Dockerfile

**Review by mspang — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

File: .github/workflows/cirque.yaml [hunk 2]
```diff
@@ -24,6 +24,17 @@ jobs:
 
         runs-on: ubuntu-latest
 
+        container:
+            image: connectedhomeip/chip-build-cirque:0.4.10
```
Comment (mingchik) [inline .github/workflows/cirque.yaml:28]: from the latest action failed message, seems like it's not able to found 0.4.10 chip-build-cirque docker image.
I guess you probably want to local generate one and then push to dockerhub.

File: .gitmodules [hunk 3]
```diff
@@ -50,6 +50,8 @@
 	path = third_party/cirque/repo
 	url = https://github.com/openweave/cirque.git
 	branch = master
+	ignore = dirty
+	commit = 905112cbeeda504223209798ec23e3754ce43fca
```
Comment (mingchik) [inline .gitmodules:54]: after you pushed 0.4.10 chip-build-cirque image to dockerhub, you might have the issue that I encountered before, which is cirque unable to pull the latest code. It keeps on using old commit id.

File: integrations/docker/images/chip-build-cirque/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    && apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -fy \
+       docker-ce docker-ce-cli containerd.io \
+    && : # aids diffs
+
+# Other Cirque prereqs
+RUN set -x \
+    && apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -fy \
+       sudo socat psmisc tigervnc-standalone-server tigervnc-viewer \
```
Comment (mingchik) [inline integrations/docker/images/chip-build-cirque/Dockerfile:35]: as I know, we always running as root in docker, do we still need to install sudo in docker container?

Comment (rwalker-apple) [reply integrations/docker/images/chip-build-cirque/Dockerfile:35]: prolly not...

File: .gitmodules (hunk 3 above)
Comment (rwalker-apple) [reply .gitmodules:54]: do you have a workaround or diagnosis of the issue you had?

File: .github/workflows/cirque.yaml (hunk 2 above)
Comment (rwalker-apple) [reply .github/workflows/cirque.yaml:28]: yeah, got ahead of myself a bit

File: integrations/docker/images/chip-build-cirque/Dockerfile (hunk 4 above)
Comment (rwalker-apple) [reply integrations/docker/images/chip-build-cirque/Dockerfile:35]: can fix later, once sudos are removed from cirque scripts

**Pull request merged**